src/compiler/checker.ts

```diff
@@ -16381,6 +16381,11 @@ namespace ts {
                 return resolveUntypedCall(node);
             }
 
+            if (isPotentiallyUncalledDecorator(node, callSignatures)) {
+                error(node, Diagnostics.This_value_has_type_0_which_accepts_too_few_arguments_to_be_used_as_a_decorator_here_Did_you_mean_to_call_it_first, typeToString(funcType));

```

Comment:
The type here seems weird. it might be long, and hides the main message. why not just say `'${0}' expects too few arguments to be called as a decorator, did you mean `@${0}()' instead?`

I think this should get the majority of the cases, for other ones we will fallback to `unable to resolve decorator when called as a function` message.